pkg/karmadactl/deinit.go

```diff
@@ -0,0 +1,345 @@
+package karmadactl
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/karmada-io/karmada/pkg/karmadactl/cmdinit/utils"
+)
+
+// LabelSelector  karmada bootstrapping label
+const LabelSelector = "karmada.io/bootstrapping"
+
+// CommandDeInitOption  options for deinit.
+type CommandDeInitOption struct {
+	Recursive       bool
+	Force           bool
+	KarmadaDataPath string
+	Namespace       string
+	KubeConfig      string
+	KubeClientSet   *kubernetes.Clientset
+}
+
+// NewCmdDeInit remove karmada on kubernetes
+func NewCmdDeInit(cmdOut io.Writer, parentCommand string) *cobra.Command {
+	opts := CommandDeInitOption{}
+	cmd := &cobra.Command{
+		Use:          "deinit",
+		Short:        "remove karmada on kubernetes",
+		Long:         "remove karmada on kubernetes",
+		Example:      deInitExample(parentCommand),
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			if err := opts.Complete(); err != nil {
+				return err
+			}
+			if err := opts.RunDeInit(); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	flags := cmd.PersistentFlags()
+	flags.StringVarP(&opts.Namespace, "namespace", "n", "karmada-system", "Kubernetes namespace")
+	flags.StringVar(&opts.KubeConfig, "kubeconfig", filepath.Join(homeDir(), ".kube", "config"), "absolute path to the kubeconfig file")
+	flags.StringVarP(&opts.KarmadaDataPath, "karmada-data", "d", "/etc/karmada", "karmada data path. kubeconfig cert and crds files")
+	flags.BoolVarP(&opts.Force, "force", "f", false, "skip delete confirmation")
+	flags.BoolVarP(&opts.Recursive, "recursive", "r", false, "delete recursively, delete all resources in the namespace, including persistent storage.")
+	return cmd
+}
+func deInitExample(parentCommand string) string {
+	return fmt.Sprintf(`
+# Remove Karmada in Kubernetes cluster.
+%s deinit
+
+# Skip the prompt and just delete Karmada.
+%s deinit --force
+
+# Recursively delete, delete the namespace and all resources under the namespace.
+%s deinit --recursive
+`, parentCommand, parentCommand, parentCommand)
```

Comment:
Just a question. What's the difference between `deinit` and `deinit --recursive`?
Does it mean the namespace will not be removed without `--recursive`?

PS: 
Haven't looked at the implementation, I guess the user might ask the same question.

Reply:
Yes, `--recursive` the namespace will be deleted. If the namespace is deleted, all resources under the namespace will be deleted.